crates/matrix-sdk/src/widget/client/implementation/outgoing.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A high-level API to generate commands (requests) that we send either
+//! directly to the widget, or to the matrix driver/client.
+
+use ruma::{
+    api::client::account::request_openid_token::v3::Response as RumaOpenIdResponse,
+    events::AnyTimelineEvent, serde::Raw, OwnedEventId,
+};
+
+use super::{
+    super::actions::{ReadEventCommand, SendEventCommand},
+    openid::OpenIdResponse,
+};
+use crate::widget::Permissions;
+
+#[async_trait::async_trait]
+pub trait RequestSender {
+    async fn send<T: Request>(&self, request: T) -> Result<T::Response, String>;
+}
```

Comment:
Unclear why this exists, can we postpone this addition, since it's used or implemented in this PR?

Reply:
Ok, removed.